awscli/customizations/eks/update_kubeconfig.py

```diff
@@ -276,10 +288,9 @@ def get_cluster_entry(self):
         the previously obtained description.
         """
 
-        cert_data = self._get_cluster_description().get("certificateAuthority",
-                                                        {"data": ""})["data"]
-        endpoint = self._get_cluster_description().get("endpoint")
-        arn = self._get_cluster_description().get("arn")
+        cert_data = self.cluster_description.get("certificateAuthority", {}).get("data", "")
+        endpoint = self.cluster_description.get("endpoint", "")
+        arn = self.cluster_description.get("arn", "")
```

Comment:
I'd prefer avoiding changing the default types here unless there is a strong reason. Before, it looks like if `endpoint` or `arn` was missing, it would return `None` but now it will be an empty string. Mainly I'm not sure how this will affect downstream logic and would opt to minimizing potential changes in values unless there is a strong reason.